src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -102,11 +102,6 @@
 
         gb.withRateLimitHandler(CUSTOMIZED);
         OkHttpClient client = new OkHttpClient().setProxy(getProxy(defaultIfBlank(apiUrl, GITHUB_URL)));
-        client.setCache(GitHubClientCacheOps.toCacheDir().apply(config));
-        if (config.getClientCacheSize() > 0) {
-            Cache cache = toCacheDir().apply(config);
-            client.setCache(cache);
-        }
```

Comment:
Yeah, I think my point 1) was just a bad test, sorry.

Given the second point and some indications that the cache is into play in another issues (returning branches even without network connectivity as @recena noted), I think it's ok to disable the cache for now.

As there are no potential compatibility issues we can activate it again later if needed and fine tune what requests need cache and which ones not.